frontend/src/features/entity_selector/AutoEntityButton.tsx

```diff
@@ -20,22 +21,43 @@ const AutoEntityButton = ({
   onResponse,
   onClick,
   disabled = false,
+  autoFill = false,
 }: Props) => {
   const { t } = useTranslation();
   const { name: pollName } = useCurrentPoll();
+  const mountedRef = useRef(false);
 
-  const askNewVideo = async () => {
+  const askNewVideo = useCallback(async () => {
     onClick();
     const newVideoId: string | null = await getVideoForComparison(
       otherUid ? idFromUid(otherUid) : null,
-      idFromUid(currentUid)
+      idFromUid(currentUid || '')
     );
+
+    if (!mountedRef.current) return;
+
     if (newVideoId) {
       onResponse(`${UID_YT_NAMESPACE}${newVideoId}`);
     } else {
       onResponse(null);
     }
-  };
+  }, [currentUid, otherUid, onClick, onResponse]);
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  const previousUidRef = useRef<string | null | undefined>(undefined);
+  useEffect(() => {
+    // We only want to autofill when the currentUid changes, not when the other dependencies change
```

Comment:
No we also want the auto fill to work when the `currentUid` is changed to `null` because that's what happens when you already are on the comparison page with videos and you click again on the "Compare" link. It may be possible to force a unmount/mount in a parent but it's probably not trivial given the current code.

I'll try to make that clearer in the comments.

Reply:
I did that in b24912325a845c39b15a77410ca96a113cfa363d.